### PR TITLE
Fix sparse_image_warp partially unknown shape

### PR DESCRIPTION
**Pull request by WindQAQ**
# Description

Fix for #2300. One leftover for old style code.

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [x] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

Passes existing and new tests.

### CODE REVIEW

**Comment by WindQAQ** [comment]
@seanpmorgan Hi Sean, please merge this before 0.12 release. Thank you!

Hi @lutzzz, can you try this to see if the issue is resolved? Thank you!

**Comment by seanpmorgan** [comment]
@lutzzz Can you please review when time allows? This is blocking our next release.

**Comment by lutzzz** [comment]
@seanpmorgan  Sorry for the delay! It is working perfectly with `Dataset.map` and `keras` models now.

**Comment by lutzzz** [comment]
Actually I am randomly seeing some `Input matrix is not invertible` errors, which are supposed to have another cause. Here is a code snippet for transforming a (image, mask) dataset with `Dataset.map(random_elastic_deformation)`

```python
def random_elastic_deformation(image: tf.Tensor, label: tf.Tensor, sigma=10, points=3, order=3) -> (tf.Tensor, tf.Tensor):
    '''Implements elastic deformation [Ronneberger et al. 2015] for a batch of images and their labels
    with bicubic (order=3) interpolation on a coarse 3 x 3 grid (points=3). The random displacement 
    vectors are sampled from a normal distribution with sigma=10.
    '''
  
    image_dtype = image.dtype
    label_dtype = label.dtype
    image  = tf.cast(image, tf.float32)
    label  = tf.cast(label, tf.float32)
    points = tf.cast(points, tf.float32)
    order  = tf.cast(order, tf.float32)
    size   = tf.cast(tf.shape(image)[1], tf.float32)

    # List of grid-like source points
    src_points = tf.stack(tf.meshgrid(tf.range(points), tf.range(points)), axis=-1)
    src_points = tf.cast(tf.reshape(src_points, (-1, 2)), tf.float32)
    src_points = src_points / points # relative coordinates
    src_points = tf.math.rint(src_points * size) # image coordinates
    src_points = tf.ones([tf.shape(image)[0], 1, 1]) * src_points

    # Destination points by random displacement vectors
    displacement = tf.random.normal(tf.shape(src_points), stddev=sigma, seed=seed)
    dst_points = tf.math.rint(src_points + displacement)

    # Calculate the flow field and use it twice
    warped_image, flow_field = tfa.image.sparse_image_warp(image, src_points, dst_points, 
        interpolation_order=order, num_boundary_points=3)
    flow_field = tf.math.rint(flow_field) # Nearest-neighbor interpolation
    warped_label = tfa.image.dense_image_warp(label, flow_field)

    warped_image = tf.cast(warped_image, image_dtype)
    warped_label = tf.cast(warped_label, label_dtype)

    return warped_image, warped_label
```

**Comment by WindQAQ** [comment]
Thanks for reviewing. That's because the random linear system you sample is not solvable. Not a bug I believe.

https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/interpolate_spline.py#L157

**Comment by lutzzz** [comment]
@WindQAQ That's what I think too. So the PR is not the source of the error and merging by @seanpmorgan would be fine.

**Review by seanpmorgan — APPROVED** [review]
Also LGTM. Thanks @WindQAQ and @lutzzz for bug report and review!
